### PR TITLE
Remove unnecessary check if label is nil in observeWithExemplar

### PR DESCRIPTION
**Pull request by dimonl**
Remove unnecessary check if label is nil in observeWithExemplar

### CODE REVIEW

File: prometheus/promhttp/instrument_server.go [hunk 1]
```diff
@@ -31,20 +31,12 @@ const magicString = "zZgWfBxLqvG8kc8IMv3POi2Bb0tZI3vAnBx+gBaFi9FyPzB/CzKUer1yufD
 // observeWithExemplar is a wrapper for [prometheus.ExemplarAdder.ExemplarObserver],
 // which falls back to [prometheus.Observer.Observe] if no labels are provided.
 func observeWithExemplar(obs prometheus.Observer, val float64, labels map[string]string) {
```
Comment (bwplotka) [inline prometheus/promhttp/instrument_server.go:33]: Nice - I would consider those functions and inline them perhaps. WDYT?

Comment (dimonl) [reply prometheus/promhttp/instrument_server.go:33]: could you please explain your idea about inline it?

Comment (bwplotka) [reply prometheus/promhttp/instrument_server.go:33]: I mean we should be able to completely remove those functions.

Comment (dimonl) [reply prometheus/promhttp/instrument_server.go]: removed it

**Review by bwplotka — APPROVED** [review]
Great, this is what I meant. Thanks <3

**Comment by bwplotka** [comment]
Do you mind signing you commit (and rebase?). Needed for DCO https://github.com/prometheus/client_golang/pull/1235/checks?check_run_id=12110217881

File: prometheus/promhttp/instrument_server.go (hunk 1 above)
Comment (dimonl) [reply prometheus/promhttp/instrument_server.go]: sure, will rebase and sign. Thanks

**Comment by dimonl** [comment]
> Do you mind signing you commit (and rebase?). Needed for DCO https://github.com/prometheus/client_golang/pull/1235/checks?check_run_id=12110217881

signed and rebased

**Comment by bwplotka** [comment]
Thanks!

**Comment by ns-rbarreda** [comment]
Unfortunately, not every `ObserverVec` implements exemplar (e.g. summary) and this change broke backwards compatibility.

The fix is in this https://github.com/prometheus/client_golang/pull/1253

**Pull request merged**